ci/build_wheel_python.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name="kvikio"
+package_dir="python/kvikio"
+
+source rapids-configure-sccache
+source rapids-date-string
+
+version=$(rapids-generate-version)
+commit=$(git rev-parse HEAD)
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+
+# This is the version of the suffix with a preceding hyphen. It's used
+# everywhere except in the final wheel name.
+PACKAGE_CUDA_SUFFIX="-${RAPIDS_PY_CUDA_SUFFIX}"
+
+pyproject_file="${package_dir}/pyproject.toml"
+
+sed -i "s/name = \"${package_name}\"/name = \"${package_name}${PACKAGE_CUDA_SUFFIX}\"/g" ${pyproject_file}
+sed -i "s/name = \"lib${package_name}\"/name = \"lib${package_name}${PACKAGE_CUDA_SUFFIX}\"/g" ${pyproject_file}
```

Comment:
Lines with `name = *` in pyproject.toml are defining the name of the current package being built. That will only occur once. This is like the package/name or outputs/name field in meta.yaml. IOW this line doesn't do anything.

What you are missing, though, is a line to modify the dependency list to account for the suffix. That's something [like this](https://github.com/rapidsai/rmm/blob/branch-24.06/ci/build_wheel_python.sh#L32).